internal/controller/apiextensions/composite/composed.go

```diff
@@ -178,28 +179,49 @@ func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Conte
 	}
 
 	for _, d := range t.ConnectionDetails {
-		if d.Name != nil && d.Value != nil {
-			conn[*d.Name] = []byte(*d.Value)
-			continue
-		}
+		switch {
+		case d.Value != nil:
+			if d.Name != nil {
+				conn[*d.Name] = []byte(*d.Value)
+			}
+		case d.FromConnectionSecretKey != nil && len(s.Data[*d.FromConnectionSecretKey]) != 0:
+			key := *d.FromConnectionSecretKey
+			if d.Name != nil {
+				key = *d.Name
+			}
 
-		if d.FromConnectionSecretKey == nil {
-			continue
+			conn[key] = s.Data[*d.FromConnectionSecretKey]
+		case d.FromFieldPath != nil && d.Name != nil:
+			_ = extractFieldPathValue(cd, d, conn)
 		}
+	}
 
-		if len(s.Data[*d.FromConnectionSecretKey]) == 0 {
-			continue
-		}
+	return conn, nil
+}
 
-		key := *d.FromConnectionSecretKey
-		if d.Name != nil {
-			key = *d.Name
-		}
+func extractFieldPathValue(from runtime.Object, detail v1.ConnectionDetail, conn managed.ConnectionDetails) error {
+	fromMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(from)
+	if err != nil {
+		return err
+	}
 
-		conn[key] = s.Data[*d.FromConnectionSecretKey]
+	str, err := fieldpath.Pave(fromMap).GetString(*detail.FromFieldPath)
+	if err == nil {
+		conn[*detail.Name] = []byte(str)
+		return nil
 	}
 
-	return conn, nil
+	in, err := fieldpath.Pave(fromMap).GetValue(*detail.FromFieldPath)
+	if err != nil {
+		return err
+	}
+
+	buffer, err := json.Marshal(in)
+	if err != nil {
+		return err
+	}
+	conn[*detail.Name] = buffer
```

Comment:
I would expect `json.Marshal` of a string value to return a byte array containing a quoted string - e.g. `[]byte("\"value\"")` - but the unit tests say otherwise. What am I missing? 🤔

Reply:
Ah - just realised this was a fallback for cases in which the value isn't a string. Should we add a test for this?